libglusterfs/src/glusterfs/common-utils.h

```diff
@@ -1245,18 +1245,18 @@ gf_tvdiff(struct timeval *start, struct timeval *end)
 
 /* Return delta value in nanoseconds. */
 
-static inline double
+static inline uint64_t
 gf_tsdiff(struct timespec *start, struct timespec *end)
 {
     struct timespec t;
 
     if (start->tv_nsec > end->tv_nsec)
-        t.tv_sec = end->tv_sec - 1, t.tv_nsec = end->tv_nsec + 1000000000;
+        t.tv_sec = end->tv_sec - 1, t.tv_nsec = end->tv_nsec + 1e9;
     else
         t.tv_sec = end->tv_sec, t.tv_nsec = end->tv_nsec;
 
-    return (double)(t.tv_sec - start->tv_sec) * 1e9 +
-           (double)(t.tv_nsec - start->tv_nsec);
+    return (uint64_t)(t.tv_sec - start->tv_sec) * 1e9 +
```

Comment:
`1e9` is a double. We should use an integer. If the raw constant can be difficult to read, we may use `GF_SEC_IN_NS` constant or similar.

Reply:
Done